### PR TITLE
refactor get non canonical blob sidecars test

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Refactor GetBlobSidecars Test: Canonical block should be a slot ahead of the fork

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java [hunk 1]
```diff
@@ -144,42 +144,35 @@ public void shouldGetNonCanonicalBlobSidecarsByRoot() throws IOException {
     chainUpdater.blockOptions.setGenerateRandomBlobs(true);
     chainUpdater.blockOptions.setGenerateRandomBlobsCount(Optional.of(4));
 
-    final UInt64 targetSlot = UInt64.valueOf(3);
-    chainUpdater.advanceChainUntil(targetSlot);
+    createBlocksAtSlots(10);
 
     final ChainBuilder fork = chainBuilder.fork();
-    SignedBlockAndState forked = fork.generateNextBlock(chainUpdater.blockOptions);
+    SignedBlockAndState forkedBlock = fork.generateNextBlock(chainUpdater.blockOptions);
```
Comment (rolfyone) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java]: `final`

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java [hunk 2]
```diff
@@ -144,42 +144,35 @@ public void shouldGetNonCanonicalBlobSidecarsByRoot() throws IOException {
...
+    final List<BlobSidecar> nonCanonicalBlobSidecars = fork.getBlobSidecars(forkedBlock.getRoot());
+    chainUpdater.saveBlock(forkedBlock, nonCanonicalBlobSidecars);
 
-    SignedBlockAndState canonical = chainBuilder.generateNextBlock(chainUpdater.blockOptions);
-    final List<BlobSidecar> canonicalBlobSidecars =
-        chainBuilder.getBlobSidecars(canonical.getRoot());
-    chainUpdater.saveBlock(canonical, canonicalBlobSidecars);
-    chainUpdater.updateBestBlock(canonical);
-    chainUpdater.finalizeEpoch(targetSlot.plus(1));
+    SignedBlockAndState canonicalBlock =
```
Comment (rolfyone) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java]: `final`

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java]: i'd probably call it a `nonCanonicalBlock` but that's very 'nit'picky...

**Review by rolfyone — APPROVED** [review]
lgtm just nits

**Review by zilm13 — APPROVED** [review]
LGTM, just finals to add that Paul mentioned

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java (hunk 1 above)
Comment (mehdi-aouadi) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java]: Done

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java (hunk 2 above)
Comment (mehdi-aouadi) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/beacon/GetBlobSidecarsIntegrationTest.java]: Done

**Pull request merged**